contrib/win32/win32compat/win32_dirent.c

```diff
@@ -62,15 +121,64 @@ DIR * opendir(const char *name)
    Return 0 if successful, -1 if not.  */
 int closedir(DIR *dirp)
 {
-    if ( dirp && (dirp->hFile) ) {
+    if (!dirp)
+        return 0;
+    if (dirp->hFile)
         _findclose( dirp->hFile );
```

Comment:
to avoid the corner cases like calling _findclose multiple times for the same hFile (or) trying to read on the hFile that is already closed.. setting the hFile to 0 inside the if loop will avoid them.